Revit_Core_Engine/Convert/Environment/FromRevit/Panel.cs

```diff
@@ -117,7 +118,13 @@ public static List<oM.Environment.Elements.Panel> EnvironmentPanelsFromRevit(thi
             if (panels != null && panels.Count != 0)
                 return panels;
 
-            Dictionary<PlanarSurface, List<PlanarSurface>> surfaces = ceiling.PanelSurfaces(ceiling.FindInserts(true, true, true, true), settings);
+            Dictionary<PlanarSurface, List<PlanarSurface>> surfaces;
+            SurfaceCache cache = refObjects.GetValue<SurfaceCache>(ceiling.Id.SurfaceCacheKey());
+            if (cache != null)
+                surfaces = cache.Surfaces;
+            else
+                surfaces = ceiling.PanelSurfaces(ceiling.FindInserts(true, true, true, true), settings);
```

Comment:
Yeah I thought about it, but would then probably create a general method for all elements, which would take `discipline` and `refObjects` as arguments, but this would then lead to confusion in case of structures that first look for outlines etc. I was afraid of creating more confusion than good by centralising this bit, so finally decided to simply leave it as is. Had a second thought about it now and would still rather keep this slight repetition at the benefit of zero confusion.